### DISCUSSION TITLE
How to control helicopter--ah1s in Unreal Engine?

### DISCUSSION THREAD

**Discussion by 140695809** [JSBSim-Team/jsbsim#1140] [Q&A] [answered]
I have seen the discussion on the AH1S. Now I know basic commands to get the AH1S flying. 
> https://github.com/JSBSim-Team/jsbsim/discussions/861#discussioncomment-5505538

I want to know how to control the attitude of ah1s in Unreal Engine, like yaw,pitch and roll.

**Comment by bcoconni** [comment]
I'm not sure if this a general question about how to pilot helicopters in general or a specific question about the Unreal engine plugin ?

**Reply by 140695809** [reply]
It is about a specific question about the Unreal engine plugin.

**Comment by gallonmate** [comment, marked as answer]
Attitude is controlled via the elevator inputs from UE5 plugin.

> For the A1HS: The collective input should be 0-1. The collective only controls the amount of lift that's generated. 0 means no lift and 1 means full lift. The other controls for the helicopter are taken from the regular aircraft inputs that control the pitch, roll, and yaw of the helicopter: Elevator input tilts the rotor blade forward/backwards which makes the helicopter tilt forwards or backwards. Aileron input tilts the rotor blade sideways which rolls the helicopter left or right. Rudder input controls the tail rotor to spin the helicopter in place left or right.
> 

If you are noticing very weak user controls, then you probably enabled the AFCS stability controls to 1, (the commands such as ap/afcs/pitch-channel-active-norm, etc). The augmented stability overrides the majority of user pilot control when it is set to 1. When you want to have stronger user control, then you reduce the AFCS, to something like 0.5. You can turn off AFCS (set to 0) when you want to make extreme maneuvers.

**Reply by 140695809** [reply]
I want to know the regular aircraft inputs that control the helicopter with JSBSim plugin in Unreal Engine.

**Reply by gallonmate** [reply]
> I want to know the regular aircraft inputs that control the helicopter with JSBSim plugin in Unreal Engine.

The information is in the Read Me page for unreal plugin here: https://github.com/JSBSim-Team/jsbsim/tree/master/UnrealEngine#key-mappings

If you are using UE5.4 and the inputs don't work for you, it's because the plugin is not yet updated to UE5.4, see this related issue. https://github.com/JSBSim-Team/jsbsim/issues/1120 You can remap the inputs yourself if you search for Unreal Tutorial on how to setup keys and inputs, etc.

**Reply by seanmcleod** [reply]
Is there a default mapping for the collective? Throttle key inputs possibly mapped to collective?

**Reply by 140695809** [reply]
I try to change the rudder data(yaw) to turn the helicopter left. When I release the yaw button, the helicopter sways left and right，out of control.
